src/client/components/dashboard/dashboard_robot/model.ts

```diff
@@ -71,9 +71,59 @@ export class DashboardRobotModel {
   @observable walkPathPlan: Vector2[]
   @observable walkCommand: Vector3
 
-  constructor(robot: RobotModel, opts: DashboardRobotModelOpts) {
+  constructor(robot: RobotModel, {
+    camera,
+    ballColor,
+    ballSightColor,
+    kickTargetColor,
+    robotColor,
+    textColor,
+    time,
+    roleName,
+    battery,
+    voltage,
+    behaviourState,
+    robotPosition,
+    robotPositionCovariance,
+    ballPosition,
+    ballCovariance,
+    kickTarget,
+    gameMode,
+    gamePhase,
+    penaltyReason,
+    playerId,
+    lastCameraImage,
+    lastSeenBall,
+    lastSeenGoal,
+    walkPathPlan,
+    walkCommand,
+  }: DashboardRobotModelOpts) {
```

Comment:
This class has a lot of dependencies! 😬 decomposing them would be a nice follow-up task.